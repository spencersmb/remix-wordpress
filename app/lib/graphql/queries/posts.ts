import gql from 'graphql-tag';
 
export const QUERY_NEXT_POSTS = gql`
    query NextPosts($after: String) {
        posts(first: 10, after: $after) {
            __typename
            pageInfo{
                endCursor
                hasNextPage
                hasPreviousPage
                startCursor
                __typename
            }
            edges {
                __typename
                node {
                    __typename
                    author {
                        node {
                            avatar {
                                height
                                url
                                width
                            }
                            id
                            name
                            slug
                        }
                    }
                    id
                    categories {
                        edges {
                            node {
                                databaseId
                                id
                                name
                                slug
                            }
                        }
                    }
                    content
                    date
                    excerpt
                    featuredImage {
                        node {
                            altText
                            caption
                            sourceUrl
                            srcSet
                            sizes
                            id
                        }
                    }
                    modified
                    databaseId
                    title
                    slug
                    isSticky
                }
            }
        }
        allSettings {
            readingSettingsPostsPerPage
        }
    }
`;

export const POST_FEATURED_IMAGE = gql`
  fragment featuredImageFields on Post {
    featuredImage {
      node {
        mediaDetails {
          width
          height
          sizes{
            width
            file
            height
            name
            sourceUrl
            mimeType
          }
        }
          altText
          caption
          sourceUrl
          srcSet
          sizes
          id
        }
      }
    }
`
export const RESOURCE_FEATURED_IMAGE = gql`
  fragment featuredImageFields__RL on ResourceLibrary {
    featuredImage {
      node {
        mediaDetails {
          sizes{
            width
            file
            height
            name
            sourceUrl
            mimeType
          }
        }
          altText
          caption
          sourceUrl
          srcSet
          sizes
          id
        }
      }
    }
`
export const RELEATED_POSTS_FIELDS = gql`
  fragment relatedPostsFields on Post {
    relatedPosts {
      ...featuredImageFields
      title
      slug
      tutorialManager {
       thumbnail {
        type
        image {
          altText
          caption
          sourceUrl
          srcSet
          sizes
          id
          mediaDetails{
            sizes{
              width
              file
              height
              name
              sourceUrl
              mimeType
            }
          }
        }
       }
      }
      categories {
        edges {
          node {
            slug
            name
          }
        }
      }
    }
  }
`
export const POST_BASIC_FIELDS = gql`
  fragment postBasicFields on Post {
    id
    content
    date
    dateGmt
    excerpt
    modified
    databaseId
    title
    slug
    isSticky
    categories {
      edges {
          node {
            databaseId
            id
            name
            slug
          }
      }
    }
    tags{
      edges{
          node{
            name
            slug
          }
      }
    }
  }
`
export const POST_RESOURCE_FIELDS = gql`
  fragment postResourceFields on Post_Tutorialmanager {
    resources {
    ... on Post_Tutorialmanager_Resources_PaidProduct {
      product {
        ... on Product {
          title
          slug
          featuredImage {
            node {
              mimeType
              mediaDetails {
                height
                width
                sizes{
                  width
                  file
                  height
                  name
                  sourceUrl
                  mimeType
                }
              }
                altText
                caption
                sourceUrl
                srcSet
                sizes
                id
            }
          }
          productDetails {
            licences {
              licenseType
              price
              url
            }
          }
        }
      }
      description
    }
    ... on Post_Tutorialmanager_Resources_Course {
      course {
        ... on Course {
          title
          slug
          link
          details {
            courseUrl
          }
          featuredImage {
            node {
              mimeType
              mediaDetails {
                height
                width
                sizes{
                  width
                  file
                  height
                  name
                  sourceUrl
                  mimeType
                }
              }
                altText
                caption
                sourceUrl
                srcSet
                sizes
                id
            }
          }
        }
      }
      description
    }
    ... on Post_Tutorialmanager_Resources_Download {
      download {
        name
        description
        url
      }
    }
    ... on Post_Tutorialmanager_Resources_ColorSwatch {
      colorSwatch {
        url
      }
    }
  }
}
`
export const PRODUCT_FIELDS = gql`
  fragment productFields on Product {
    title 
    slug
    featuredImage {
      node {
        mimeType
        mediaDetails {
          height
          width
          sizes{
            width
            file
            height
            name
            sourceUrl
            mimeType
          }
        }
          altText
          caption
          sourceUrl
          srcSet
          sizes
          id
      }
    }
    productDetails{
      licences {
        licenseType
        price
        url
      }
    }
  }
`
export const QUERY_POST_BY_ID = gql`
    ${POST_BASIC_FIELDS}
    ${POST_FEATURED_IMAGE}
    ${RELEATED_POSTS_FIELDS}
    query postById($id: ID!) {
        post(idType: DATABASE_ID, id: $id) {
          ...postBasicFields
          ...featuredImageFields
          ...relatedPostsFields
          seo{
              title
              opengraphPublishedTime
              opengraphModifiedTime
              metaDesc
              readingTime
          }
          author {
              node {
                  avatar {
                      height
                      url
                      width
                  }
                  id
                  name
                  slug
                  uri
              }
          }
          etSocialNav{
              pinterestMeta {
                description
              }
              pinterestImage{
                altText
                caption
                sourceUrl
                srcSet
                sizes
                id
                mediaDetails{
                  sizes{
                    width
                    file
                    height
                    name
                    sourceUrl
                    mimeType
                  }
                }
              }
          }
          tutorialManager {
              postExcerpt
              thumbnail {
                image {
                  altText
                  caption
                  sourceUrl
                  srcSet
                  sizes
                  id
                  mediaDetails{
                    width
                    height
                    sizes{
                      width
                      file
                      height
                      name
                      sourceUrl
                      mimeType
                    }
                  }
                }
              }
              youtube {
                id
                addVideoMetadata
                duration
                videoObject {
                  description
                  uploadDate
                  thumbnailUrl
                  clipElements {
                    name
                    startOffset
                    endOffset
                  }
                  potentialActions{
                    name
                    startOffset
                  }
                }
              }
              downloads {
                  ... on ResourceLibrary {
                      title
                      freebie {
                          downloadLink
                      }
                  }
              }
          }
          comments(first: 500, after: null, where: {parent: null}) {
              pageInfo{
                endCursor
                hasNextPage
              }
              edges {
                  node {
                      databaseId
                      approved
                      parent {
                        node {
                          databaseId
                        }
                      }
                      id
                      author {
                          node {
                            isRestricted
                              name
                              ... on CommentAuthor {
                                gravatar{
                                  url
                                }
                              }
                          }
                      }
                      date
                      commentedOn {
                          node {
                              id
                              status
                          }
                      }
                      content
                      replies {
                          edges {
                              node {
                                  id
                                  databaseId
                                  content
                                  date
                                  parent {
                                    node {
                                      databaseId
                                    }
                                  }
                                  author {
                                      node {
                                          name
                                        ... on CommentAuthor {
                                          gravatar{
                                            url

                                          }
                                        }
                                      }
                                  }
                                  
                              }
                          }
                      }
                  }
              }
          }
        }
    }
  `