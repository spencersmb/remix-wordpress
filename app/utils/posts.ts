import { isEmpty } from "lodash";

export function flattenAllPosts(posts:any): IPost[] | false{
  const postsFiltered = posts?.edges?.map(({ node = {} }) => node);
  return Array.isArray(postsFiltered) && postsFiltered.map(mapPostData)
}

export function rearrangeLicenses(licenses: ILicense[]){
  return licenses.reduce((acc: any, licence) => {
        // Add it to the beginning of the array
        if(licence.licenseType === 'standard'){
          acc.unshift(licence)
        }
        console.log('licence.licenseType', licence.licenseType);
        
        // if there is only one item in array, add extended as the 2nd
        if(licence.licenseType === 'extended'){
          acc.push(licence)
        }

        // if the ext is the last item to get looped over
        if(licence.licenseType === 'extended' && acc.length === 2){
          const standard = acc[0]
          const server = acc[1]
          acc = [
            standard,
            licence,
            server
          ]
        }

        if(licence.licenseType === 'server'){
          acc.push(licence)
        }

        return acc
      }, [])
}

export function mapPostData(post:IPostRaw | {} = {}): IPost {
  const data = { ...post };
  let modifiedData: any = {...post}

  // Clean up the author object to avoid someone having to look an extra
  // level deeper into the node
  if (data.author)  {
    modifiedData.author = {
      ...data.author.node,
    }
  }

  // Clean up the categories to make them more easy to access
  if (data.categories) {
    modifiedData.categories = data.categories.edges.map(({ node }) => {
      return {
        ...node,
      };
    });
  }

  // Clean up the featured image to make them more easy to access
  if (data.featuredImage) {
    modifiedData.featuredImage = data.featuredImage.node;
  }

  if (data.tags) {
    modifiedData.tags = data.tags.edges.map(({ node }) => {
      return {
        name: node.name,
      };
    });
  }

  if(data.relatedPosts){
    modifiedData.relatedPosts = data.relatedPosts.map((post: IPostRaw) => {
      return mapPostData(post)
    })
  }

  // if(data.downloadManager?.downloads){
  //   modifiedData.downloadManager = data.downloadManager?.downloads.map(download => {
  //     return download.downloadDetails
  //   })
  // }

  if(data.tutorialManager?.paidProducts){
    const newPaidProducts = data.tutorialManager?.paidProducts.map(product => {
      const newProduct = {
        ...product,
        details: {
          ...product.details,
          licences: product.details.licences ? rearrangeLicenses(product.details.licences) : null,
        }
      }
      return newProduct
    })
    modifiedData.tutorialManager = {
      ...data.tutorialManager,
      paidProducts: newPaidProducts
    }
  }


  if(data.comments){
    modifiedData.comments = {
      pageInfo: data.comments.pageInfo,
      list: data.comments.edges.map(({ node }) => {
        return parseComment(node)
      }),
    }

  }

  return modifiedData

}

export function parseComment(node: IPostCommentRaw): IPostComment {

  return {
    ...node,
    replies: node.replies.edges.map(({ node }) => {
      return {
        ...node,
        parent: node.parent?.node.databaseId || null,
        author: node.author.node
      }
    }),
    parent: node.parent?.node.databaseId || null,
    author: node.author.node
  }
}

interface IMediaDetailSize {
  width: string
  file: string
  height: string
  name: string
  sourceUrl: string
}
export function getMediaSizeUrl(mediaSizes: IMediaDetailSize[] | undefined, name: string): IMediaDetailSize{
  if(mediaSizes === undefined || mediaSizes?.length < 1) return  {
    width: '',
    file: '',
    height: '',
    name: '',
    sourceUrl: ''
  }
  
  return mediaSizes.reduce((previousValue: any, currentValue: any) => {

    if(currentValue.name === name){
      return currentValue
    }else{ 
      return previousValue
    }

  }, {})
}

export function formatDate(date: string): string{
  const blogDate = new Date(date)
  const monthIndex: number = blogDate.getMonth()
  const day: number = blogDate.getDate()
  
  const months: string[] = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December'
  ]
  return `${months[monthIndex]} ${day}, ${blogDate.getFullYear()}`
}

export function splitProgramNameInTitle(title: string):{title: string, subTitle: string | undefined} {
  let newTitle = title
  switch (Boolean(title)) {

    case title.includes("in Procreate"):
      newTitle = title.replace("in Procreate", "")
      return {
        title: newTitle,
        subTitle: "in Procreate",
      }
      // return (
      //   <>
      //     <div className='mb-3'>{newTitle}</div>
      //     <div className={`${splitCss}`}>in Procreate</div>
      //   </>
      // )

    case title.includes("in Illustrator"):
      newTitle = title.replace("in Illustrator", "")
      return{
        title: newTitle,
        subTitle: "in Illustrator",
      }
      // return (
      //   <>
      //     <div className='mb-3'>{newTitle}</div>
      //     <div className={`${splitCss}`}>in Illustrator</div>
      //   </>
      // )
    case title.includes("in Adobe Illustrator"):
      newTitle = title.replace("in Adobe Illustrator", "")
      return {
        title: newTitle,
        subTitle: "in Adobe Illustrator",
      }
      // return (
      //   <>
      //     <div className='mb-3'>{newTitle}</div>
      //     <div className={`${splitCss}`}>in Adobe Illustrator</div>
      //   </>
      // )
    case title.includes("in InDesign"):
      newTitle = title.replace("in InDesign", "")
      return {
        title: newTitle,
        subTitle: "in InDesign",
      }
      // return (
      //   <>
      //     <div className='mb-3'>{newTitle}</div>
      //     <div className={`${splitCss}`}>in InDesign</div>
      //   </>
      // )
    case title.includes("in Photoshop"):
      newTitle = title.replace("in Photoshop", "")
      return {
        title: newTitle,
        subTitle: "in Photoshop",
      }
    default:
      return {
        title: newTitle,
        subTitle: undefined,
      }
  }
}

export function findString(str: string, arr: string[]) {
  return arr.find(item => item.includes(str))
}

export function findSkillLevel(categories: ICategories[]): ICategories | undefined {
  let tutorialsFound = false
  let defaultSkill = {
    name: "beginner",
  }
  let skillFoundStopLooking = false
  const skillFound = categories.reduce((previousValue: any, currentValue: ICategories):any => {
    if(currentValue.slug === "tutorials"){
      tutorialsFound = true
    }
    if(skillFoundStopLooking){
      return previousValue
    }
    switch(currentValue.slug ){
      
      case "advanced":
        skillFoundStopLooking = true
        return currentValue
      case "intermediate":
        skillFoundStopLooking = true
        return currentValue
      case "beginner":
        skillFoundStopLooking = true
        return currentValue
      default:
        return previousValue
    }
  }, {})

    return !isEmpty(skillFound) 
    ? skillFound 
    : isEmpty(skillFound) && tutorialsFound 
      ? defaultSkill
      : undefined
}

export function getLicense(licenses: ILicense[] | null, type: LicenseEnum) {
  if (!licenses) {
    return null
  }
  return licenses.reduce((acc, curr) => {
    if (curr.licenseType === type) {
      return curr
    }
    return acc
  }, {
    licenseType: '',
    price: 0,
    url: '',
  })
}