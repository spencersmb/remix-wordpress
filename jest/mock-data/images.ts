 export const mockFeaturedImage: IFeaturedImage = {
    altText: '',
    sourceUrl: '',
    mediaDetails: {
      sizes: [],
      height: 0,
      width: 0,
    },
    id: '',
    mimeType: '',
    sizes: '',
    srcSet: '',
  }
  export const mockFeatureImageComplete: IFeaturedImage = {
      mediaDetails: {
        height: 0,
        width:  0,
        sizes: [
          {
            width: "1000",
            file: "create-candy-cane-lettering-in-procreate-hero-1000x749.jpg",
            height: "749",
            name: "headless_post_thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1000x749.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "600",
            file: "create-candy-cane-lettering-in-procreate-hero-600x360.jpg",
            height: "360",
            name: "headless_resource_image",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-600x360.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "20",
            file: "create-candy-cane-lettering-in-procreate-hero-20x20.jpg",
            height: "20",
            name: "placeholder",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-20x20.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1000",
            file: "create-candy-cane-lettering-in-procreate-hero-1000x749.jpg",
            height: "749",
            name: "headless_ipad",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1000x749.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1440",
            file: "create-candy-cane-lettering-in-procreate-hero-1440x810.jpg",
            height: "810",
            name: "headless_post_feature_image",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1440x810.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "150",
            file: "create-candy-cane-lettering-in-procreate-hero-150x150.jpg",
            height: "150",
            name: "thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-150x150.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "500",
            file: "create-candy-cane-lettering-in-procreate-hero-500x281.jpg",
            height: "281",
            name: "medium",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-500x281.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1024",
            file: "create-candy-cane-lettering-in-procreate-hero-1024x576.jpg",
            height: "576",
            name: "large",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1024x576.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1536",
            file: "create-candy-cane-lettering-in-procreate-hero-1536x864.jpg",
            height: "864",
            name: "1536x1536",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1536x864.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "100",
            file: "create-candy-cane-lettering-in-procreate-hero-100x56.jpg",
            height: "56",
            name: "et_thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-100x56.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "550",
            file: "create-candy-cane-lettering-in-procreate-hero-550x550.jpg",
            height: "550",
            name: "readanddigest_square",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-550x550.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "800",
            file: "create-candy-cane-lettering-in-procreate-hero-800x600.jpg",
            height: "600",
            name: "readanddigest_landscape",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-800x600.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "600",
            file: "create-candy-cane-lettering-in-procreate-hero-600x800.jpg",
            height: "800",
            name: "readanddigest_portrait",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-600x800.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1200",
            file: "create-candy-cane-lettering-in-procreate-hero-1200x675.jpg",
            height: "675",
            name: "readanddigest_post_feature_image",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1200x675.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "128",
            file: "create-candy-cane-lettering-in-procreate-hero-128x86.jpg",
            height: "86",
            name: "readanddigest_thumb",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-128x86.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1200",
            file: "create-candy-cane-lettering-in-procreate-hero-1200x580.jpg",
            height: "580",
            name: "readanddigest_single_post_title",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1200x580.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "1100",
            file: "create-candy-cane-lettering-in-procreate-hero-1100x550.jpg",
            height: "550",
            name: "readanddigest_large_width",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1100x550.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "550",
            file: "create-candy-cane-lettering-in-procreate-hero-550x1080.jpg",
            height: "1080",
            name: "readanddigest_large_height",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-550x1080.jpg",
            mimeType: "image/jpeg"
          },
          {
            width: "421",
            file: "create-candy-cane-lettering-in-procreate-hero-421x203.jpg",
            height: "203",
            name: "wp_rp_thumbnail",
            sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-421x203.jpg",
            mimeType: "image/jpeg"
          }
        ]
      },
      mimeType: "image/jpeg",
      altText: "Create Candy Cane Lettering in Procreate",
      sourceUrl: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero.jpg",
      srcSet: "https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-500x281.jpg 500w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1440x810.jpg 1440w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1024x576.jpg 1024w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1536x864.jpg 1536w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-100x56.jpg 100w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero-1200x675.jpg 1200w, https://etheadless.local/wp-content/uploads/2021/11/create-candy-cane-lettering-in-procreate-hero.jpg 1920w",
      sizes: "(max-width: 500px) 100vw, 500px",
      id: "cG9zdDoxMDA4NQ=="
  }