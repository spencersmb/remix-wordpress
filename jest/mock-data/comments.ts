import { mockAuthorData } from "./posts"

export const mockComment: IPostComment = {
  databaseId: 218427,
  // approved: true,
  parent: null,
  id: "Y29tbWVudDoyMTg0Mjc=",
  author: {
    id: "YWRtaW5pc3RyYXRvcjox",
    databaseId: "1",
    name: "Spencer Bigum",
    gravatar: {
      url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
    }
  },
  date: "2022-01-27 21:08:02",
  content: "<p>Comment Body</p>\n",
  replies: []
}

export const mockComments: IComments = {
    pageInfo: {
    endCursor: "YXJyYXljb25uZWN0aW9uOjIxODQyNA==",
    hasNextPage: true
  },
  list: [
    {
      databaseId: 218427,
      // approved: true,
      parent: null,
      id: "Y29tbWVudDoyMTg0Mjc=",
      author: {
        id: "YWRtaW5pc3RyYXRvcjox",
        databaseId: "1",
        name: "Spencer Bigum",
        gravatar: {
          url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
        }
      },
      date: "2022-01-27 21:08:02",
      // commentedOn: {
      //   node: {
      //     id: "cG9zdDo1NDM0",
      //     status: "publish"
      //   }
      // },
      content: "<p>4</p>\n",
      replies: []
      },
    {
      databaseId: 218426,
      // approved: true,
      parent: null,
      id: "Y29tbWVudDoyMTg0MjY=",
      author: {
        databaseId: "1",
        id: "YWRtaW5pc3RyYXRvcjox",
        name: "Spencer Bigum",
        gravatar: {
          url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
        }
      },
      date: "2022-01-27 21:07:22",
      // commentedOn: {
      //   node: {
      //     id: "cG9zdDo1NDM0",
      //     status: "publish"
      //   }
      // },
      content: "<p>3</p>\n",
      replies: [
          {
            id: "Y29tbWVudDoyMTg0MzM=",
            databaseId: 218433,
            content: "<p>two test</p>\n",
            date: "2022-01-28 01:31:37",
            parent: 218426,
            author: {
              databaseId: "2",
              id: "YWRtaW5pc3RyYXRvcjoxMg==",
              name: "twotest",
              gravatar: {
                url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
              }
            }
          },
          {
            id: "Y29tbWVudDoyMTg0MzI=",
            databaseId: 218432,
            content: "<p>test</p>\n",
            date: "2022-01-28 01:26:12",
            parent: 218426,
            author: {
              databaseId: "1",
              id: "YWRtaW5pc3RyYXRvcjox",
              name: "Spencer M Bigum",
              gravatar: {
              url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
              }
            }
            },
          {
            id: "Y29tbWVudDoyMTg0Mjg=",
            databaseId: 218428,
            content: "<p>tes</p>\n",
            date: "2022-01-27 21:08:26",
            parent: 218426,
            author: {
              databaseId: "3",
              id: "YWRtaW5pc3RyYXRvcjoxMw==",
              name: "Teela Bigum",
              gravatar: {
                url: "https://secure.gravatar.com/avatar/aac786a8b226660a1064233ededdda6b?s=96&d=mm&r=g"
              }
            }
          }
        ]
      },
    {
      databaseId: 218424,
      // approved: true,
      parent: null,
      id: "Y29tbWVudDoyMTg0MjQ=",
      author: {
        databaseId: "1",
        id: "YWRtaW5pc3RyYXRvcjox",
        name: "Spencer",
        gravatar: {
          url: "https://secure.gravatar.com/avatar/9119dc766dbfe9f6b05b7f39f9dbd9ce?s=96&d=mm&r=g"
        }
      },
      date: "2022-01-27 21:05:42",
      // commentedOn: {
      //   node: {
      //     id: "cG9zdDo1NDM0",
      //     status: "publish"
      //   }
      // },
      content: "<p>test</p>\n",
      replies: []
      }
  ]
}

export const mockPostCommentRaw: IPostCommentRaw = {
  author: {
    node: {
      ...mockAuthorData,
      databaseId: '1',
      id: '1',
      gravatar: {
        url: 'https://www.gravatar.com/avatar/00000000000000000000000000000000?d=mm&s=24',
      }
    }
  },
  content: "This is a comment",
  databaseId: 1,
  approved: false,
  parent: null,
  commentedOn:{
    node:{
      databaseId: 1
    }
  },
  date: "2018-04-23T15:00:00",
  id: '1',
  replies: {
    edges:[
      {
        node: {
          author: {node: {
            ...mockAuthorData,
            databaseId: '1',
            id: '1',
            gravatar: {
              url: 'https://www.gravatar.com/avatar/00000000000000000000000000000000?d=mm&s=24',
            }
          }},
          content: 'comment reply',
          databaseId: 10,
          date: '2018-04-23T15:00:00',
          id: '10',
          parent:{
            node: {
              databaseId: 1
            }
          }
        }
      }
    ]
  }
}