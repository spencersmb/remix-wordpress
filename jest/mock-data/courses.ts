import { mockFeatureImageComplete } from "./images";

export const mockCourseRaw:ICourseRaw = {
  id: 1,
  details:{
    courseTags: [
      {
        tag: 'tag1'
      }
    ],
    courseUrl: 'https://www.example.com'
  },
  slug: 'course-slug',
  title: 'Course Title',
  featuredImage: {
    node: {
      ...mockFeatureImageComplete
    }
  }
}

export const mockCourse: ICourse = {
  id: 1,
  details:{
    courseTags: ['tag1'],
    courseUrl: 'https://www.example.com'
  },
  slug: 'course-slug',
  title: 'Course Title',
  featuredImage: mockFeatureImageComplete
}

export const mockCourse2: ICourse = {
  id: 2,
  details:{
    courseTags: ['tag2'],
    courseUrl: 'https://www.example2.com'
  },
  slug: 'course2-slug',
  title: 'Course2 Title',
  featuredImage: mockFeatureImageComplete
}

export const mockCourse3: ICourse = {
  id: 3,
  details:{
    courseTags: ['tag3'],
    courseUrl: 'https://www.example3.com'
  },
  slug: 'course3-slug',
  title: 'Course3 Title',
  featuredImage: mockFeatureImageComplete
}