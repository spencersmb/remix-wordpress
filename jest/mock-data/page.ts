import { mockFeaturedImage } from "./images";
import { mockAuthorData, mockSeo } from "./posts";

export const mockPage: IPage = {
  seo: {
    title: "Page Title",
    metaDesc: "Page description",
    opengraphModifiedTime: "2020-01-01T00:00:00.000Z",
    opengraphPublishedTime: "2020-01-01T00:00:00.000Z",
    readingTime: 1
  },
  title: "Page Title",
  slug: "page-slug",
  content: "Page Content",
  author: mockAuthorData,
  featuredImage: mockFeaturedImage,
  date: "2020-01-01",
  id: "page-id",
}