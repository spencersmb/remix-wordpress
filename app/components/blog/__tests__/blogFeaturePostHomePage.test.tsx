import { mockPostData } from '@TestUtils/mock-data/posts';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import BlogFeaturePostHomePage from "../blogFeaturePostHomePage";

it('Should render Blog Feature Post Home Page Component ', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <BlogFeaturePostHomePage post={mockPostData} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
})