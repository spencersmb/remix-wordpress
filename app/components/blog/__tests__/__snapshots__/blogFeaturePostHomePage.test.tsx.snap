// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Blog Feature Post Home Page Component  1`] = `
<div
  className="flex flex-col tablet:flex-row tablet:gap-4 tablet:items-start laptop:flex-row-reverse laptop:items-center laptop:justify-end"
>
  <div
    className="relative max-w-[450px] w-full px-4 mb-[68px] tablet:max-w-[300px] tablet:mb-0 tablet:mt-8 laptop:w-[50%] desktop:max-w-[480px]"
  >
    <div
      className="z-2 absolute top-[-48px] left-[45%] -translate-x-1/2 w-[95px] desktop:top-[-50px]"
    >
      <div
        className="relative flex-1"
        data-testid="imigx-padding-bot"
        style={
          Object {
            "paddingBottom": "78.5997357992074%",
          }
        }
      >
        <div
          className="absolute w-full lazy-load-wrapper lazy-load-wrapper-block lazy-load-image-full transition-all duration-300"
          data-testid="imigx-container"
        >
          <span
            className=" lazy-load-image-background blur"
            style={
              Object {
                "backgroundImage": "url(https://et-website.imgix.net/et-website/images/black-pin-min.png?auto=format&w=20&fit=clip)",
                "backgroundSize": "100% 100%",
                "color": "transparent",
                "display": "inline-block",
                "height": "595px",
                "width": "757px",
              }
            }
          >
            <span
              className=""
              style={
                Object {
                  "display": "inline-block",
                  "height": "595px",
                  "width": "757px",
                }
              }
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="p-3 bg-white shadow-md -rotate-3 tablet:p-3 laptop:p-4"
      data-testid="polaroid-container"
    >
      <div>
        <div
          className="relative flex-1"
          data-testid="padding-bot"
          style={
            Object {
              "paddingBottom": "88.8%",
            }
          }
        >
          <div
            className="absolute w-full lazy-load-wrapper lazy-load-wrapper-block lazy-load-image-full"
          >
            <img
              alt="Every Tuesday Fallback Featured Thumbnail"
              aria-label="Product Image"
              data-testid="feature-image"
              height="888px"
              onLoad={[Function]}
              sizes=""
              src="https://et-website.imgix.net/defaultImages/default-thumb.jpg"
              srcSet=""
              width="1000px"
            />
          </div>
        </div>
      </div>
      <div
        className="relative"
      >
        <div
          className="pb-[18%]"
        />
      </div>
    </div>
  </div>
  <div
    className="flex flex-col items-start flex-1 mb-12 tablet:mb-0 tablet:mt-14 laptop:max-w-[430px] laptop:w-full"
  >
    <a
      className="group"
      href="/test-slug"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
    >
      <h2
        className="relative text-3xl transition-all duration-300 font-sentinel__SemiBoldItal tablet:text-4xl desktop:text-5xl laptop:group-hover:text-sage-500"
      >
        <div
          className="!top-[-50px] left-[-15px] tablet:!top-[-63px] text-grey-800 laptop:group-hover:text-grey-800 text-4xl absolute font-bonVivant text-2xl tablet:text-5xl italic top-[-47px] left-[-25px] rotate-[-14deg]"
        >
          Latest Post
        </div>
        test-title
      </h2>
    </a>
    <div
      className="flex flex-row gap-8 mt-4 text-sm desktop:text-base"
    >
      <div>
        undefined NaN, NaN
      </div>
    </div>
    <div
      className="mt-4 desktop:text-lg"
      dangerouslySetInnerHTML={
        Object {
          "__html": "EXCERPT",
        }
      }
    />
    <a
      className="mt-8 btn btn-xl btn-primary tablet:mt-6 tablet:btn-flex"
      href="/test-slug"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
    >
      View the post
    </a>
  </div>
</div>
`;
